server/controllers/users/userController.ts

```diff
@@ -0,0 +1,78 @@
+import { Response } from 'express';
+import Interviewee from '../../database/Models/Interviewee';
+import { RequestType } from '../../utils';
+import validateQuery from '../../utils/validation/queryValidation';
+
+const getAllReviews = async (req: RequestType, res: Response) => {
+  const { page = '1', saved = undefined } = req.query;
+
+  await validateQuery(req.query);
+  // Convert the incoming saved string into a boolean
+  const savedBoolean : any = saved ? JSON.parse(saved.toString())
+    : saved;
+
+  // const id = req.userInfo?.id;
+
+  // filter reviews based on the saved parameter and paginate the results
+  // TODO: Replace the static userId with the logged in user's id when create review is implemented
+  let filteredReviews : any | undefined;
+  if (savedBoolean === true || savedBoolean === false) {
+    filteredReviews = await Interviewee.aggregate([{
+      $project: {
+        _id: 0,
+        userId: 1,
+        'interviews.review': 1,
+      },
+    },
+
+    {
+      $unwind: '$interviews',
+    },
+    {
+      $match: {
+        userId: '4',
+        'interviews.review.saved': savedBoolean,
+      },
+    },
+    {
+      $group: {
+        _id: '$interviews.review._id',
+        review: {
+          $first: '$interviews.review',
+        },
+      },
+    },
+    ]).skip((Number(page) - 1) * 3).limit(3);
+  }
+
+  if (savedBoolean === undefined) {
```

Comment:
Add the condition only before the filter line so you don't end up with duplicated code

Reply:
I can't add if conditions into mongoose object :/

Reply:
I still think we can edit this without duplication, you can conditionally change just the value you are using in the filter.
When it is undefined, then it could be one of these values true or false.

Reply:
Check this option when connecting to the database `ignoreUndefined`